namer/namer.cc

```diff
@@ -1414,6 +1414,57 @@ class SymbolDefiner {
         }
     }
 
+    void deleteViaFullNameHash(core::MutableContext ctx, const core::FoundFieldHash &oldDefHash) {
+        auto ownerRef = core::FoundDefinitionRef(core::FoundDefinitionRef::Kind::Class, oldDefHash.owner.idx);
+        ENFORCE(oldDefHash.nameHash.isDefined(), "Can't delete rename if old hash is not defined");
+
+        // Because a change to classes would have take the slow path, should be safe
+        // to look up old owner in current foundDefs.
+        auto ownerSymbol = getOwnerSymbol(ownerRef);
+        ENFORCE(ownerSymbol.isClassOrModule());
+        auto owner = ownerSymbol.asClassOrModuleRef();
+        if (oldDefHash.owner.onSingletonClass) {
+            owner = owner.data(ctx)->singletonClass(ctx);
+        }
+
+        // We have to accumulate a list of fields to delete, instead of deleting them in the loop
+        // below, because deleting a field invalidates the members() iterator.
+        vector<core::FieldRef> toDelete;
+
+        // Note: this loop is accidentally quadratic. We run deleteViaFullNameHash once per field
+        // previously defined in this file, then in each call look at each of that field's class's members
```

Comment:
lol what was I saying here
```suggestion
        // previously defined in this file, then in each call look at each member of that field's owner
```

if you want i'd appreciate if you changed the other method too 🙈

Reply:
#6339 for the method version, will fix the field version here.